File: app/src/main/java/com/lttrung/dormitory/ui/viewwaterbills/ViewWaterBillsUseCase.java
package com.lttrung.dormitory.ui.viewwaterbills;

import java.lang.reflect.Type;

import javax.inject.Singleton;

import io.reactivex.rxjava3.core.Single;

@Singleton
public interface ViewWaterBillsUseCase {
    // Bổ sung params, thay đổi Type bằng class tương ứng
    Single<Type> getWaterBills();
}

File: app/src/main/java/com/lttrung/dormitory/ui/main/home/viewwaterbills/ViewWaterBillsUseCase.kt
package com.lttrung.dormitory.ui.main.home.viewwaterbills

import com.lttrung.dormitory.database.data.network.models.WaterBill
import io.reactivex.rxjava3.core.Single
import javax.inject.Singleton

@Singleton
interface ViewWaterBillsUseCase {
    fun getWaterBills(): Single<List<WaterBill>>
}